jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -372,6 +381,113 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
         managerState.flags[StateFlags.IS_ONGOING] = false
     }
 
+    /**
+    helper function to generate 1p dependency metrics
+    makes GET requests to maven central to check for 1p dependencies for a small subset of the project's
+    dependencies
+     */
+    private fun makeGETRequests(groupId: String, artifactId: String, version: String, jobId: JobId): Boolean {
+        val httpClient = HttpClient.newHttpClient();
+        val request =  HttpRequest.newBuilder()
+            .uri(URI.create("https://search.maven.org/solrsearch/select?q=g:$groupId%20AND%20a:$artifactId%20AND%20v:$version%20AND%20p:jar&rows=20&wt=json"))
+            .GET()
+            .build();
+
+        try{
+            val response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+
+            if(response.statusCode() == 200){
+                val jsonResponse = response.body();
+                try {
+                    val json: JsonObject = Json.parseToJsonElement(jsonResponse) as JsonObject;
+                    val responses = json["response"] as? JsonObject;
+                    val numFound = responses?.get("numFound");
+
+
+                    if(numFound.toString() == "0"){
+                        CodetransformTelemetry.projectContains1pDependencies(
+                            codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+                            codeTransformJobId = jobId.id,
+                            has1pdependency = true
+                        )
+                        return true;
+                    }
+                } catch (e: Exception){
+                    LOG.warn{"Error parsing JSON response from GET request to maven central: ${e.printStackTrace()}"}
+                }
+            } else {
+                LOG.warn{"Error making request to maven central. Response code: ${response.statusCode()}"}
+            }
+        } catch(e: Exception){
+            LOG.warn{"Exception while making GET request to maven central: ${e.printStackTrace()}"}
+        }
+        return false;
+    }
+
+
+    /**
+     * Coroutine to find and generate metrics on 1p dependencies (dependencies
+     * not available in maven central)
+     */
+    fun generate1pDependencyMetrics(jobId: JobId, session: CodeModernizerSession) = projectCoroutineScope(project).launch {
```

Comment:
there should definitely be a way to get this as structured output instead of scraping HTML, which tends to be fragile